### DISCUSSION TITLE
Chromecast support

### DISCUSSION THREAD

**Discussion by jankaifer** [snapcast/snapcast#1235] [General]
Original issue with motivation is here: https://github.com/badaix/snapcast/issues/750

I'm creating this as a sub-issue to debug and investigate snapcast via chromecast.

**Comment by jankaifer** [comment]
There are a bunch of relevant messages in the original thread: https://github.com/badaix/snapcast/issues/750

**Comment by jankaifer** [comment]
I do have 3 different devices that I'm testing it on:
1. Basic Chromecast (that thing with HDMI output)
2. Google Nest Audio speaker
3. Acer Speaker Halo

**Comment by jankaifer** [comment]
When I tested a naive implementation where I just created an example receiver that redirects directly to the snapweb hosted by snapcast server It behaved differently on each device:

1. It was very laggy on the basic chromecast, The sound was delayed by several seconds and it the audio was slower than normal and very choppy

2. It works fine on Google Next Audio, it is only a little bit delayed compared to using snapweb on my laptop (it works fine after setting the latency to ~120ms in the snapweb).

3. It does not start at all on the Acer Halo speaker for some reason :disappointed:

**Comment by jankaifer** [comment]
Chromecast is too wonky and I decided I do not care enough to dig deeper. If you want to continue here you can use [my fork](https://github.com/jankaifer/snapweb/tree/jankaifer/chrome-cast-support-2) of `snapweb` with needed changes.

Then you need to create and deploy your own receiver (this should be just blank page with som JS that redirects the chromecast to snapweb url, chromecast will load even http:// page using this trick).

**Comment by badaix** [comment]
I will not add official support for chrome cast devices, as I don't own any and I don't know if there is an API available.
I will turn this into a discussion, maybe it helps someone.
